networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRangeMessageHandler.java

```diff
@@ -181,5 +189,18 @@ void incrementCurrentSlot() {
       remainingBlocks = remainingBlocks.minus(ONE);
       currentSlot = currentSlot.plus(step);
     }
+
+    SafeFuture<Optional<SignedBeaconBlock>> loadNextBlock() {
+      final UnsignedLong slot = this.currentSlot;
+      final Bytes32 knownBlockRoot = knownBlockRoots.get(slot);
+      if (knownBlockRoot != null) {
+        return combinedChainDataClient
+            .getBlockByBlockRoot(knownBlockRoot)
+            .thenApply(maybeBlock -> maybeBlock.filter(block -> block.getSlot().equals(slot)));
+      } else {
```

Comment:
I think you need to check if the slot is less than the smallest slot in `knownBlockRoots`, otherwise you'll run extra lookups for skipped slots.

Reply:
Good spot.  Added a test to confirm empty slots are not included in results from `ForkChoiceUtil.getAncestorRoots` as well. Wound up tweaking the handling for requests that extend beyond our best block as well.